src/app/modules/form/public-form/__tests__/public-form.controller.spec.ts

```diff
@@ -1488,4 +1490,335 @@ describe('public-form.controller', () => {
+      // Arrange
+      const MOCK_FORM = {
+        authType: AuthType.NIL,
+        esrvcId: '12345',
+      } as IFormSchema
+      const mockRes = expressHandler.mockResponse()
+      const expectedError = {
+        message:
+          'Please ensure that the form has authentication enabled. Please refresh and try again.',
+      }
+      MockFormService.retrieveFormById.mockReturnValueOnce(okAsync(MOCK_FORM))
+
+      // Act
+      await PublicFormController.handleValidateFormEsrvcId(
+        MOCK_REQ,
+        mockRes,
+        jest.fn(),
+      )
+
+      // Assert
+      expect(mockRes.status).toBeCalledWith(400)
+      expect(mockRes.json).toBeCalledWith(expectedError)
+    })
+
+    it('should return 400 when the form has authType.CP', async () => {
+      // Arrange
+      const MOCK_FORM = {
+        authType: AuthType.CP,
+        esrvcId: '12345',
+      } as IFormSchema
+      const mockRes = expressHandler.mockResponse()
+      const expectedError = {
+        message:
+          'Please ensure that the form has authentication enabled. Please refresh and try again.',
+      }
+      MockFormService.retrieveFormById.mockReturnValueOnce(okAsync(MOCK_FORM))
+
+      // Act
+      await PublicFormController.handleValidateFormEsrvcId(
+        MOCK_REQ,
+        mockRes,
+        jest.fn(),
+      )
+
+      // Assert
+      expect(mockRes.status).toBeCalledWith(400)
+      expect(mockRes.json).toBeCalledWith(expectedError)
+    })
+
+    it('should return 400 when the form does not have an eServiceId', async () => {
+      // Arrange
+      const MOCK_FORM = {
+        authType: AuthType.SP,
+      } as SpcpForm<IFormSchema>
+      const mockRes = expressHandler.mockResponse()
+      const expectedError = {
+        message:
+          'This form does not have a valid eServiceId. Please refresh and try again.',
+      }
+      MockFormService.retrieveFormById.mockReturnValueOnce(okAsync(MOCK_FORM))
+
+      // Act
+      await PublicFormController.handleValidateFormEsrvcId(
+        MOCK_REQ,
+        mockRes,
+        jest.fn(),
+      )
+
+      // Assert
+      expect(mockRes.status).toBeCalledWith(400)
+      expect(mockRes.json).toBeCalledWith(expectedError)
+    })
+
+    it('should return 404 when there is no form with the given formId', async () => {
+      // Arrange
+      const mockRes = expressHandler.mockResponse()
+      const expectedError = {
+        message:
+          'Could not find the form requested. Please refresh and try again.',
+      }
+      MockFormService.retrieveFormById.mockReturnValueOnce(
+        errAsync(new FormNotFoundError()),
+      )
+
+      // Act
+      await PublicFormController.handleValidateFormEsrvcId(
+        MOCK_REQ,
+        mockRes,
+        jest.fn(),
+      )
+
+      // Assert
+      expect(mockRes.status).toBeCalledWith(404)
+      expect(mockRes.json).toBeCalledWith(expectedError)
+    })
+
+    it('should return 500 when the returned html has no title', async () => {
```

Comment:
actually I think we shouldn't be returning a 5xx here, but let's preserve existing behaviour for now
```suggestion
    it('should return 502 when the returned html has no title', async () => {
```